Ceen.Httpd/ResponseOutputStream.cs

```diff
@@ -155,23 +158,23 @@ public override async Task WriteAsync(byte[] buffer, int offset, int count, Canc
 
 			if (!m_passThrough)
 			{
-				if (m_response.HasSentHeaders || m_response.ContentLength >= 0 || ((m_buffer == null ? 0 : m_buffer.Length) + count) > MAX_RESPONSE_BUFFER)
+				if (m_response.HasSentHeaders || m_response.ContentLength >= 0 || m_buffer?.Length + count > MAX_RESPONSE_BUFFER)
+				{
 					m_passThrough = true;
+				}
 				else
 				{
 					if (m_buffer == null)
 						m_buffer = new MemoryStream();
 
 					await m_buffer.WriteAsync(buffer, offset, count, cancellationToken);
 					m_written += count;
-

```

Comment:
This fix is certainly correct. The data has been added to the buffer, so the code cannot continue. An alternative would be to set `count = 0;` so line 176 does not write the data again.

Reply:
count=0 won't prevent SetLengthAndFlushAsync, which will break buffering logic.
Also it looks like a hack :)

Reply:
It _should_ flush if the maximum amount of buffer space has been used. But I am good with returning.

Reply:
If there won't be enough space in buffer, codition on line 161 will be met and buffer will be flushed.